### PR TITLE
--txpool.gossip.disable

### PR DESCRIPTION
**Pull request by AskAlexSharov**

### CODE REVIEW

File: cmd/rpcdaemon/cli/httpcfg/http_cfg.go [hunk 1]
```diff
@@ -79,4 +79,6 @@ type HttpCfg struct {
 	MaxGetProofRewindBlockCount int  //Max GetProof rewind block count
 	// Ots API
 	OtsMaxPageSize uint64
+
+	DisableTxPoolGossip bool
```
[the remark this answers is not in the document]
Comment (AskAlexSharov) [reply cmd/rpcdaemon/cli/httpcfg/http_cfg.go]: fixed

File: erigon-lib/txpool/pool.go [hunk 2]
```diff
@@ -2126,6 +2139,20 @@ func (p *TxPool) deprecatedForEach(_ context.Context, f func(rlp []byte, sender
 	})
 }
 
+// disables adding remote transactions
+type TxPoolDropRemote struct {
+	*TxPool
+}
+
+func NewTxPoolDropRemote(txPool *TxPool) *TxPoolDropRemote {
```
[the remark this answers is not in the document]
Comment (AskAlexSharov) [reply erigon-lib/txpool/pool.go]: removed

File: erigon-lib/txpool/txpooluitl/all_components.go [hunk 3]
```diff
@@ -143,16 +143,16 @@ func AllComponents(ctx context.Context, cfg txpoolcfg.Config, cache kvcache.Cach
 		cancunTime = cfg.OverrideCancunTime
 	}
 
-	txPool, err := txpool.New(newTxs, chainDB, cfg, cache, *chainID, shanghaiTime, agraBlock, cancunTime, maxBlobsPerBlock, logger)
+	pool, err := txpool.New(newTxs, chainDB, cfg, cache, *chainID, shanghaiTime, agraBlock, cancunTime, maxBlobsPerBlock, logger)
```
[the remark this answers is not in the document]
Comment (AskAlexSharov) [reply erigon-lib/txpool/txpooluitl/all_components.go]: fixed

**Comment by Boyuan-Chen** [comment]
@AskAlexSharov do you want me to fix the unit tests?

**Comment by AskAlexSharov** [comment]
@Boyuan-Chen thank you. tests seems green - just merge-conflict. fixed.

**Pull request merged**